src/psyclone/dynamo0p3.py

```diff
@@ -3503,6 +3507,7 @@ def __init__(self, node):
                         self._cma_ops[arg.name]["intent"] = arg.intent
                         self._cma_ops[arg.name]["datatype"] = \
                             arg.intrinsic_type
+                        self._cma_ops[arg.name]["kind"] = arg.precision
```

Comment:
Again, not my code (I think it's Andy's). I think this approach was chosen because the information about intent, datatype and precision is required for both PSy-layer and kernel-stub generation so it made sense to have a table in the `__init__` method to be used for both (CMA data matrix is required in both cases). Fields and LMA operators have clearer delineation between the PSy-layer and the kernel-stub code.